### PR TITLE
Make miri read_dir test a little more robust

### PR DESCRIPTION
**Pull request by SUPERCILEX**
r? @RalfJung

### CODE REVIEW

File: src/tools/miri/tests/pass-dep/shims/fs.rs [hunk 1]
```diff
@@ -394,23 +395,27 @@ fn test_directory() {
...
             .unwrap()
-            .map(|e| e.unwrap().metadata().unwrap().is_file())
-            .collect::<Vec<_>>()
+            .map(|e| {
+                let e = e.unwrap();
+                (e.file_name(), e.metadata().unwrap().is_file())
+            })
+            .collect::<HashMap<_, _>>()
     );
     // Deleting the directory should fail, since it is not empty.
```
[the remark this answers is not in the document]
Comment (SUPERCILEX) [reply src/tools/miri/tests/pass-dep/shims/fs.rs:420]: Yup, thanks. The tests are failing to compile with a tokio error locally, so this going to be a little painful, sorry.

[the remark this answers is not in the document]
Comment (RalfJung) [reply src/tools/miri/tests/pass-dep/shims/fs.rs:420]: Could you open an issue with the command you are using to run the tests, and the error that shows?

They should work locally, too.

[the remark this answers is not in the document]
Comment (SUPERCILEX) [reply src/tools/miri/tests/pass-dep/shims/fs.rs:420]: https://github.com/rust-lang/rust/issues/103273

**Pull request merged**